File: citizen_science/lib/dto/auth_response.dart
import 'user_response.dart';

/// Data transfer object for authentication responses.
/// 
/// Contains the JWT token and user information returned after
/// successful login or registration.
class AuthResponse {
  final String token;
  final UserResponse user;

  AuthResponse({
    required this.token,
    required this.user,
  });

  /// Creates an [AuthResponse] from a JSON map.
  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      token: json['token'],
      user: UserResponse.fromJson(json['user']),
    );
  }

  /// Converts this [AuthResponse] to a JSON map.
  Map<String, dynamic> toJson() {
    return {
      'token': token,
      'user': user.toJson(),
    };
  }
}

File: CitizenScienceBackend/src/main/java/com/citizenScience/dto/AuthResponse.java
package com.citizenscience.dto;

import com.citizenscience.entities.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Data Transfer Object for authentication responses.
 * Contains JWT token and user information.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AuthResponse {
    private String token;
    private UserResponse user;

    /**
     * Creates an AuthResponse from token and user entity.
     * 
     * @param token the JWT token
     * @param user the user entity
     * @return AuthResponse instance
     */
    public static AuthResponse from(String token, User user) {
        return AuthResponse.builder()
                .token(token)
                .user(UserResponse.from(user))
                .build();
    }
}
